src/Maestro/tests/Maestro.ScenarioTests/EndToEndFlowLogic.cs

```diff
@@ -0,0 +1,352 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.Maestro.Client.Models;
+using Microsoft.Extensions.Primitives;
+using NUnit.Framework;
+
+namespace Maestro.ScenarioTests
+{
+    [TestFixture]
+    public class EndToEndFlowLogic : MaestroScenarioTestBase
+    {
+        private readonly string testRepo1Name = "maestro-test1";
```

Comment:
Is it worth extracting some of these constants to a common utility class?

Reply:
If these repos are a real, external resource, that makes sense, since all the tests need to use the same 3 values.  Anything that's just fake data, I would leave, since it would be fine for one test to have a different value somewhere.

We could call it something like "TestRepository" or something, and it could contain any values/helper methods related to interacting with them.

Reply:
They're real repos and the coherency commits/branches matter, the rest is arbitrary (branches, commits, etc). I'll go ahead and pull the real stuff out to a utility class.